### PR TITLE
fix(codegen): resolve template dependencies recursively

### PR DESCRIPTION
**Pull request by StarpTech**
#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

We forgot to collect all template dependencies recursively. I run into an issue that specifying the `fastifyServer` template was not enough because it has template dependencies that requires other templates. 

```ts
	codeGenerators: [
		{
			templates: [templates.typescript.client, templates.typescript.fastifyServer],
			path: '../../web/src/generated',
		},
	],
```

I also added a max template depth to protect against the infinite loop.

### CODE REVIEW

File: packages/sdk/src/codegen/index.ts [hunk 1]
```diff
@@ -37,33 +37,31 @@ const ensurePath = (filePath: string) => {
 	}
 };
 
-export const doNotEditHeader = '// Code generated by wunderctl. DO NOT EDIT.\n\n';
+export const collectAllTemplates = (templates: Template[], level = 0) => {
+	if (level > 10) {
+		throw new Error('Template dependency loop detected');
```
Comment (jensneuse) [inline packages/sdk/src/codegen/index.ts:42]: Does it make sense to throw? or just stop? Recursive dependencies might happen, but shouldn't be a problem I'd say.

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: packages/sdk/src/codegen/index.ts (hunk 1 above)
Comment (StarpTech) [reply packages/sdk/src/codegen/index.ts:42]: Definitely, if you have more than 10 levels, you are definitely doing something wrong.

Comment (StarpTech) [reply packages/sdk/src/codegen/index.ts:42]: We can also return it and log an error when a specific depth has been reached.

**Review by jensneuse — APPROVED** [review]

**Comment by jensneuse** [comment]
LGTM

**Pull request merged**